### PR TITLE
ddns use interface, select ip_source

### PR DESCRIPTION
**Pull request by livepu**
(cherry picked from commit 8efce52c52c406b10a6e589966d3fc04b173d690)

# Conflicts:
#	package/plugin-gargoyle-ddns/files/www/templates/ddns_service_template

### CODE REVIEW

File: package/plugin-gargoyle-ddns/files/www/js/ddns.js [hunk 1]
```diff
@@ -424,7 +424,14 @@ function setUciFromDocument(dstUci, pkg, section)
 	dstUci.set("ddns_gargoyle", section, "", "service");
 	dstUci.set("ddns_gargoyle", section, "enabled", "1");
 	dstUci.set("ddns_gargoyle", section, "service_provider", providerName);
-	dstUci.set("ddns_gargoyle", section, "ip_source", "internet");
+	var ip_from=document.getElementById("ip_from").value;//select or  input? input is easy
```
Comment (lantis1008) [inline package/plugin-gargoyle-ddns/files/www/js/ddns.js:427]: Please cleanup comments

File: package/plugin-gargoyle-ddns/files/www/js/ddns.js [hunk 2]
```diff
@@ -424,7 +424,14 @@ function setUciFromDocument(dstUci, pkg, section)
...
+	if(ip_from!="internet"){
```
Comment (lantis1008) [inline package/plugin-gargoyle-ddns/files/www/js/ddns.js:428]: Please use standard Gargoyle coding style (braces get their own line)

File: package/plugin-gargoyle-ddns/files/www/js/ddns.js [hunk 3]
```diff
@@ -535,6 +542,20 @@ function setDocumentFromUci(srcUci, pkg, section)
 	var forceDays = (getMultipleFromUnit( srcUci.get("ddns_gargoyle", section, "force_unit") ) * srcUci.get("ddns_gargoyle", section, "force_interval"))/(24*60*60);
 	forceDays = (forceDays > 0) ? forceDays : 3;
 	document.getElementById( "ddns_force" ).value = forceDays;
+	
+	var ip_from=srcUci.get("ddns_gargoyle",section,"ip_interface");//2020-03-21
+	var selectip = document.getElementById("ip_from");
+	if(ip_from){
```
Comment (lantis1008) [inline package/plugin-gargoyle-ddns/files/www/js/ddns.js:548]: Please replace this section with setSelectedValue or srtSelectedText (whichever is most appropriate)

File: package/plugin-gargoyle-ddns/files/www/templates/ddns_service_template [hunk 4]
```diff
@@ -21,7 +21,19 @@
 						<em><%~ days %></em>
 					</span>
 				</div>
-
+				<div class="row form-group">
+					<label class="col-xs-5" for="ip_from" id="ip_from_label">ip_from:</label>
```
Comment (lantis1008) [inline package/plugin-gargoyle-ddns/files/www/templates/ddns_service_template:25]: Missing i18n translations

File: package/plugin-gargoyle-ddns/files/www/templates/ddns_service_template [hunk 5]
```diff
@@ -21,7 +21,19 @@
...
+					<span class="col-xs-7">
+						<select id="ip_from" name="ip_from">
+						<option value="internet">internet</option>
```
Comment (lantis1008) [inline package/plugin-gargoyle-ddns/files/www/templates/ddns_service_template:28]: Please fix indentation in this area

**Review by lantis1008 — CHANGES_REQUESTED** [review]
Thanks for this. Please see individual feedback items. 
Also @ericpaulbishop do you have any objections? Are you still working on ddns? Do not want to conflict your changes.

**Comment by livepu** [comment]
Sorry, I didn't understand this standard before. Or I need you to deal with it. I don't really know how to use GitHub.

It was a test at the beginning, and it was stable after a period of time. Recently, someone mentioned that DDNS is unstable, so this code is issued.

**Comment by lantis1008** [comment]
I will fix your code in this instance, once Eric has had an opportunity to respond. 
For what it's worth, this doesn't fix ddns. For some people this option may be more appropriate. For most users, it is not needed. Issues reported for earlier versions of 1.12 around the aruljohn site have been fixed. If people won't use the latest versions, how is your fix going to help them either lol?
However, you have gone to the effort of exposing the option, so I will include it. I appreciate your efforts, thank you.

**Pull request closed without merging**